styx-client/src/main/java/com/spotify/styx/client/StyxApolloClient.java

```diff
@@ -452,7 +457,8 @@ private Request decorateRequest(
             return response;
           default:
             final String message = response.status().code() + " " + response.status().reasonPhrase();
-            throw new ApiErrorException(message, response.status().code(), authToken.isPresent());
+            final String responseRequestId = response.headers().getOrDefault(X_REQUEST_ID, requestId);
```

Comment:
If there's a nonnull ID in headers() different from requestId, then that seems like a "should never happen" kind of situation that would be good to highlight in error log, not just shrug about it.

Reply:
Should that fail the request or should it just result in an angry log message?

Reply:
No very strong opinion. A loud fail seems :+1: